Public/Src/Tools/SymbolDaemon/VsoSymbolClient.cs

```diff
@@ -0,0 +1,208 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.ContractsLight;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using BuildXL.Ipc.Common;
+using BuildXL.Ipc.Interfaces;
+using Microsoft.VisualStudio.Services.Common;
+using Microsoft.VisualStudio.Services.Content.Common;
+using Microsoft.VisualStudio.Services.Content.Common.Authentication;
+using Microsoft.VisualStudio.Services.Content.Common.Tracing;
+using Microsoft.VisualStudio.Services.Symbol.App.Core;
+using Microsoft.VisualStudio.Services.Symbol.App.Core.Telemetry;
+using Microsoft.VisualStudio.Services.Symbol.App.Core.Tracing;
+using Microsoft.VisualStudio.Services.Symbol.Common;
+using Microsoft.VisualStudio.Services.Symbol.WebApi;
+using Newtonsoft.Json;
+using static BuildXL.Utilities.FormattableStringEx;
+
+namespace Tool.SymbolDaemon
+{
+    /// <summary>
+    /// Client for Artifact Services symbols endpoint.
+    /// </summary>
+    public sealed class VsoSymbolClient : ISymbolClient
+    {
+        private const DebugEntryCreateBehavior DefaultDebugEntryCreateBehavior = DebugEntryCreateBehavior.ThrowIfExists;
+
+        private static IAppTraceSource Tracer => SymbolAppTraceSource.SingleInstance;
+
+        private readonly ILogger m_logger;
+        private readonly SymbolConfig m_config;
+        private readonly ISymbolServiceClient m_symbolClient;
+        private readonly CancellationTokenSource m_cancellationSource;
+        private CancellationToken Token => m_cancellationSource.Token;

```

Comment:
Nit: I think we generally use full name CancellationToken or just short ct. I haven't seen 'Token' used....